### DISCUSSION TITLE
Runtime environment info

### DISCUSSION THREAD

**Discussion by NightProgramming** [RebeccaStevens/deepmerge-ts#43] [Q&A] [answered]
Can you please give some info about the supported runtime environments? Is Node required? Can the library be used in a browser (if so, what ES version do browsers need to support)?  Can polyfills be used in order to use this library on old browsers (e.g. IE11)? Which polyfills would be required (I see things like Object.getOwnPropertySymbols being used)?

**Comment by RebeccaStevens** [comment, marked as answer]
This library isn't zero-dependency, hence node is needed to load dependencies and thus it cannot be used in browsers out of the box. Out of the box, the library uses ES2018.

That being said, the library uses no node specific apis and thus can be used in browsers fairly easily. You'll just need to use a build tool (such as [webpack](https://webpack.js.org/) or [rollup](https://rollupjs.org/)) to bundle the library for browsers. These bundlers (with the use of plugins such as [babel](https://babeljs.io/)) will be able to handle down leveling JavaScript and adding any needed polyfills.

**Reply by NightProgramming** [reply]
Awesome, thank you!

**Comment by RebeccaStevens** [comment]
Alternatively, you can use a CDN like [skypack.dev](https://www.skypack.dev/view/deepmerge-ts) to provide a web-ready version of the library.
